src/main/java/frc/utils/joysticks/SmartJoystick.java

```diff
@@ -0,0 +1,212 @@
+package frc.utils.joysticks;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import edu.wpi.first.wpilibj2.command.button.POVButton;
+
+public class SmartJoystick {
+
+    public final JoystickButton A, B, X, Y, L1, R1, START, BACK, L3, R3;
+
+    public final AxisButton L2, R2;
+
+    public final POVButton POV_UP, POV_RIGHT, POV_DOWN, POV_LEFT;
+
+    private final double deadzone;
+
+    private Joystick joystick;
+
+    /**
+     * This constructor constructs the joystick based on the joystick port we give it.
+     *
+     * @param joystick_port The port of the joystick.
+     * @param deadzone values if the stick below it will count as 0.
+     */
+    public SmartJoystick(int joystick_port, double deadzone) {
+        this(new Joystick(joystick_port), deadzone);
+    }
+
+    /**
+     * This constructor uses a joystick and assigns it button values and numbers.
+     *
+     * @param stick The joystick object.
+     * @param deadzone values if the stick below it will count as 0.
+     */
+    public SmartJoystick(Joystick stick, double deadzone) {
+        this.deadzone = deadzone;
+        this.joystick = stick;
+        this.A = new JoystickButton(joystick, 1);
+        this.B = new JoystickButton(joystick, 2);
+        this.X = new JoystickButton(joystick, 3);
+        this.Y = new JoystickButton(joystick, 4);
+        this.L1 = new JoystickButton(joystick, 5);
+        this.R1 = new JoystickButton(joystick, 6);
+        this.BACK = new JoystickButton(joystick, 7);
+        this.START = new JoystickButton(joystick, 8);
+        this.L3 = new JoystickButton(joystick, 9);
+        this.R3 = new JoystickButton(joystick, 10);
+        this.L2 = new AxisButton(joystick, 2, 0.1);
+        this.R2 = new AxisButton(joystick, 3, 0.1);
+        this.POV_UP = new POVButton(joystick, 0);
+        this.POV_RIGHT = new POVButton(joystick, 90);
+        this.POV_DOWN = new POVButton(joystick, 180);
+        this.POV_LEFT = new POVButton(joystick, 270);
```

Comment:
So much numbers
![image](https://github.com/GreenBlitz/GB-Robot-Template/assets/26098399/8136de12-0edd-4b74-b8ae-6f583dbcfb81)

Reply:
It’s the button port, make constants for each one is kind of over kill
Ignore r2 l2 I will delete them

Reply:
So create an enum